torch/distributed/nn/jit/instantiator.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python3
+import importlib
+import inspect
+import os
+import pathlib
+
+import torch
+from torch.distributed.nn.jit.code_template import CodeTemplate
+from torch.distributed.nn.jit.templates import TEMPLATES_DIR_PATH
+from torch.distributed.nn.jit.templates.instantiated import INSTANTIATED_TEMPLATE_DIR_PATH
+
+
+def infer_module_interface_cls(module_creator, module_interface_cls):
+    # If module_interface_cls is provided, use it.
+    if module_interface_cls is not None:
+        return module_interface_cls
+
+    # Otherwise, if module_creator is a class ctor, use it.
+    if inspect.isclass(module_creator):
+        return module_creator
```

Comment:
Shouldn't we also check that the class is also either of type nn.Module or ScriptModule?

Reply:
Good point! 
Should be nn.Module or RecursiveScriptModule.